addons/website/__manifest__.py

```diff
@@ -271,6 +274,9 @@
             'website/static/src/xml/website.cookies_bar.xml',
             'website/static/src/js/editor/commands_overridden.js',
             'website/static/src/js/editor/odoo_editor.js',
+            'website/static/src/components/edit_head_body_dialog/edit_head_body_dialog.js',
+            'website/static/src/components/edit_head_body_dialog/edit_head_body_dialog.scss',
+            'website/static/src/components/edit_head_body_dialog/edit_head_body_dialog.xml',
```

Comment:
> For that, I had to import the edit_head_body_dialog component in website.assets_wysiwyg (it was only in website.assets_editor, I don't think there is a section that combines both?).

If you are in edit mode, you are in the backend and if you are in the backend, `website.assets_editor` is loaded (it is directly included in the backend assets). So I am not sure why this would be necessary?

Reply:
I didn't anticipate it either, but without it, on opening the editor, we get this error:

```
The following modules are needed by other modules but have not been defined, they may not be present in the correct asset bundle:

    @website/components/edit_head_body_dialog/edit_head_body_dialog

The following modules could not be loaded because they have unmet dependencies, this is a secondary error which is likely caused by one of the above problems:

    @website/snippets/s_embed_code/options
```

The `s_embed_code/options` are imported within `website.assets_wysiwyg` (like all snippets options).

Besides, there are a few other common lines between `website.assets_wysiwyg` and `website.assets_editor`:
```
            ('include', 'web._assets_helpers'),
            'web/static/src/scss/pre_variables.scss',
            'web/static/lib/bootstrap/scss/_variables.scss',
```

So I'd assume this is the right way.

Reply:
> The s_embed_code/options are imported within website.assets_wysiwyg (like all snippets options).

Yes this is the lazy loaded bundle that we load when entering edit mode. But now that I think about it, I wonder if it is not also injected inside the iframe for reasons I don't remember. And inside the iframe, there is no backend assets, which could be the issue?

> Besides, there are a few other common lines between website.assets_wysiwyg and website.assets_editor:
> So I'd assume this is the right way.

But those lines are SCSS files which only contains functions and variables. They do not produce code that would be loaded and executed twice.

Reply:
`<t t-call-assets="website.assets_wysiwyg" t-js="false"/>` in the iframefallback. There is probably no reason for that :grimacing: -> to investigate (maybe speak with ARD, he remembers maybe more than me: I don't see where we inject that bundle in the base iframe anymore, maybe we removed it? If that's the case, it's probably still a mistake to have it in the fallback iframe).

Reply:
As a summary of what the code was supposed to do:
- Some JS was needed as the editor changes prototypes of HTMLElements, ([like done here](https://github.com/odoo/odoo/blob/6ae1b7af7ff4553a9f06e59340db1ae9339d0b84/addons/web_editor/static/src/js/editor/odoo-editor/src/commands/deleteForward.js#L75)), and that needs to be loaded in the iframe too (but not in the iframefallback).
- In the iframefallback, we wanted to make sure that any css that impacts the main iframe, also impacts the fallback, so as a precaution, we also loaded assets_wysiwyg. This is most likely not needed, or should not be needed.
- I tried to change it with [this PR](https://github.com/odoo/odoo/pull/111258) but in the end, solution was not clean and I could not find of a way to do it that was cleaner, so I put it aside.
- One of the complexity (at the time) was that we cannot load two bundles which contains some of the same files. That means loading only a few files then a full version, or a full version right away would require more code than just changing the manifests. (Which I think was the reason we put it aside)

@qsm-odoo 
After checking the code, it seems like bundles were heavily changed by https://github.com/odoo/odoo/commit/fac09ae974ebb7bc552ac0b9f59836b4b89e9e34 , so it might be worth to re-investigate what is loaded inside the iframe (which is done [here](https://github.com/odoo/odoo/blob/6ae1b7af7ff4553a9f06e59340db1ae9339d0b84/addons/website/static/src/js/content/website_root_instance.js#L11)) 

I suggested to @robinlej to move the code for `s_embed/options.js` into the new `website.backend_assets_wysiwyg` bundle as to avoid importing any new code into the iframe. It seems to work, maybe we could move all of the website options code in that bundle too?

Reply:
That's what I kinda remembered indeed. We can probably just change `s_embed/options.js` in 17.0 and in master, move everything at the right place :+1: